disco/storage/outputs.py

```diff
@@ -0,0 +1,250 @@
+
+    def __init__(self, output: pathlib.Path) -> None:
+        self.output = self.validate_output(output)
+
+    @property
+    @abstractmethod
+    def output_type(self):
+        """The output type"""
+
+    @property
+    def creation_time(self):
+        timestamp = get_creation_time(self.output)
+        return datetime.fromtimestamp(timestamp)
+
+    @property
+    def config_file(self):
+        return self.output / "config.json"
+
+    @property
+    def result_file(self):
+       return self.output / "results.json"
+
+    @property
+    def job_outputs(self):
+        return self.output / "job-outputs"
+    
+    @property
+    def table_names(self):
+        return TABLE_NAMES
+    
+    @property
+    def feeder_head_table(self):
+        return self.output / TABLE_NAMES[0]
+    
+    @property
+    def feeder_losses_table(self):
+        return self.output / TABLE_NAMES[1]
+    
+    @property
+    def metadata_table(self):
+        return self.output / TABLE_NAMES[2]
+    
+    @property
+    def thermal_metrics_table(self):
+        return self.output / TABLE_NAMES[3]
+    
+    @property
+    def voltage_metrics_table(self):
+        return self.output / TABLE_NAMES[4]
+
+    @property
+    def snapshot_time_points_table(self):
+        return self.output / TABLE_NAMES[5]
+
+    @property
+    def hosting_capacity_results(self):
+        return self.output.glob("hosting_capacity_*.json")
+
+    def __str__(self):
+        """Return string representation of the output instance"""
+        return str(self.output)
+
+    @abstractmethod
+    def validate_output(self, output):
+        """Given an output, check if it contains desired reports
+        
+        Parameters
+        ----------
+        output: pathlib.Path
+
+        Returns
+        -------
+        str, the valid output directory containing tables.
+        """
+
+
+class SimulationOutput(OutputBase):
+    """Output instance class for handling Jade output"""
+    
+    @property
+    def output_type(self):
+        return OutputType.JADE
+    
+    def validate_output(self, output):
+        """Check if output containing all desired reports
+        
+        Parameters
+        ----------
+        output: pathlib.Path
+        """
+        report_files = [output / table_name for table_name in self.table_names[:-1]]
+        report_exists = [report_file.exists() for report_file in report_files]
+        if not all(report_exists):
+            raise ValueError(f"The output '{output}' does not contain valid reports.")
+        return output
+
+
+class PipelineSimulationOutput(OutputBase):
+    """Output class for handling Jade pipeline output"""
+    
+    @property
+    def output_type(self):
+        return OutputType.JADE_PIPELINE
+    
+    def validate_output(self, output):
+        """Return stage output that contains desired reports
+        
+        Parameters
+        ----------
+        output: str or pathlib.Path
```

Comment:
Looks like it always returns Path

Reply:
It's about the input parameters, could be str or Path.